sync/src/synchronizer/block_fetcher.rs

```diff
@@ -171,11 +166,7 @@ impl<'a> BlockFetcher<'a> {
                 header = self
                     .synchronizer
                     .shared
-                    .get_header_view(
-                        &parent_hash,
-                        Some(status.contains(BlockStatus::BLOCK_STORED)),
-                    )?
-                    .into_inner();
+                    .get_header_index_view(&parent_hash, false)?;
```

Comment:
we have checked the status if contains `BLOCK_STORED` in line 148 and returned early, no need to check it again here.